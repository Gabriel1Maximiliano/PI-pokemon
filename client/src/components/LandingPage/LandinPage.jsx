import React from 'react'
import { NavLink } from 'react-router-dom'

export const LandinPage = () => {
  return (
    <div className='landing-container' >
      <div className='landing-title' >
      <h1 id='h1-bienvenida' >Bienvenidos a mi p√°gina de Pokemon</h1>
      </div>
      
       <NavLink className='nav-link-landing' to='/home'>
        <button className='btn-landing'>Ingresar</button>
      </NavLink> 
     
    </div>
  )
}
