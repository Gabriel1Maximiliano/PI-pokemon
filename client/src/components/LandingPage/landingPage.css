.landing-container{
    margin-left: 5px;
    width: 97vw;
      height: 93vh;
    background-color: #F70245;
    border: solid 10px;
  
    background-image: url('./images/turma-pokemon-png.png');
  }
  .btn-landing:hover{
    background-color: coral;
    transition: .5s;
    color:black;
   
    
  }
  
  .btn-landing{
   justify-content: center;
    display: flex;
    flex-direction: column;
    margin-left: auto;
    margin-top: 450px;
    margin-right: auto;
    font-family: Verdana, Geneva, Tahoma, sans-serif;
    font-weight: 1000;
    height: 50px;
    width: 100px;
    border-radius: 12px;
    padding-left: 15px;
    text-align: center;
    
    
  }
  .landing-title{
    display: flex;
    margin-bottom: 12px;
    justify-content: center;
    margin-left: 5px;
    
  color: aliceblue;
    
   
  }
  
  .nav-link-landing{
    
    color: #ABABA9;
    text-decoration: none;
  }