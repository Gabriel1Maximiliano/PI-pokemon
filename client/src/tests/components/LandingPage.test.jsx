import { fireEvent, render, screen } from "@testing-library/react";
import {LandinPage} from '../../components/LandingPage/LandinPage'
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event'

describe('Pruebas sobre LandingPage ', () => { 
    test('Debe hacer match con el Snapshot', () => { 
      const { container }= render( 
       <BrowserRouter>
       <LandinPage />
       </BrowserRouter>)
      expect( container ).toMatchSnapshot()
      
     });

     test('debe mostrar el título "Bienvenidos a mi página de Pokemon" ' ,() => {
               render( 
            <BrowserRouter>
            <LandinPage />
            </BrowserRouter>);
           expect( screen.getByRole('heading',{Name: / Bienvenidos a mi página de Pokemon/}) )
     })
     test('Debe contener un boton con el nombre "Ingresar"', () => { 
        render( 
            <BrowserRouter>
            <LandinPage />
            </BrowserRouter>);

            expect( screen.getByRole('button',{Name : /Ingresar/}) )
          
      })
     
 })
