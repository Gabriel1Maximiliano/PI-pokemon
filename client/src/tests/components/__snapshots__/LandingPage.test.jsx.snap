// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre LandingPage  Debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="landing-container"
  >
    <div
      class="landing-title"
    >
      <h1
        id="h1-bienvenida"
      >
        Bienvenidos a mi p√°gina de Pokemon
      </h1>
    </div>
    <a
      class="nav-link-landing"
      href="/home"
    >
      <button
        class="btn-landing"
      >
        Ingresar
      </button>
    </a>
  </div>
</div>
`;
